docs/source/howto/codes.rst

```diff
@@ -45,8 +45,90 @@ Design guidelines
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+As an example, we will show how to use the ``arithmetic.add`` plugin, which is a pre-installed plugin implemented in `bash` to sum two integers.
+You can access it with the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+Next, we provide the inputs for the code when running the calculation.
+Use ``verdi plugin`` to determine what inputs a specific plugin expects:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
+    (...)
+        Inputs:
+               code:  required  Code        The `Code` to use for this job.
+                  x:  required  Int, Float  The left operand.
+                  y:  required  Int, Float  The right operand.
+    (...)
+
+You will see that 3 inputs nodes are required: two containing the values to add up (``x``, ``y``) and one containing information about the specific code to execute (``code``).
+If you already have these nodes in your database, you can get them by :ref:`querying for them <how-to:data:finding-data>` or using ``orm.load_node(<PK>)``.
+Otherwise, you will need to create them as shown below (note that you `will` need to already have the ``localhost`` computer configured):
+
+.. code-block:: python
+
+    from aiida import orm
+    bash_binary = orm.Code(remote_computer_exec=[localhost, '/bin/bash'])
+    number_x = orm.Int(17)
+    number_y = orm.Int(11)
+
+To provide these as inputs to the calculations, we will now use the ``builder`` object that we can get from the class:
+
+.. code-block:: python
+
+    calculation_builder = calculation_class.get_builder()
+    calculation_builder.code = bash_binary
+    calculation_builder.x = number_x
+    calculation_builder.y = number_y
+
+Now everything is in place and ready to perform the calculation, which can be done in two different ways.
+The first one is blocking and will return a dictionary containing all the output nodes (keyed after their label, so in this case these should be: "remote_folder", "retrieved" and "sum") that you can safely inspect and work with:
+
+.. code-block:: python
+
+    from aiida.engine import run
+    output_dict = run(calculation_builder)
+    sum_result = output_dict['sum']
+
+The second one is non blocking, as you will be submitting it to the daemon and control is immediately returned to the interpreter.
+The return value in this case is the calculation node that is stored in the database.
+
+.. code-block:: python
+
+    from aiida.engine import submit
+    calculation = submit(calculation_builder)
+
+Note that, although you have access to the node, the underlying calculation `process` is not guaranteed to have finished when you get back control in the interpreter.
+In order to keep track of it you can use the verdi command line interface:
+
+.. code-block:: bash
+
+    $ verdi process list
```

Comment:
Maybe a link here to the topic on monitoring processes: https://aiida.readthedocs.io/projects/aiida-core/en/docs-revamp/topics/processes/usage.html#monitoring-processes